src/Mod/Arch/ArchRoof.py

```diff
@@ -286,393 +383,455 @@ def makeRoofProfilsDic(self, id, angle, run, idrel, overhang, thickness,):
-            rampantNext2 = DraftGeomUtils.edg(points[0],points[1])
-            point = DraftGeomUtils.findIntersection(rampantCurrent,rampantNext2,infinite1=True,infinite2=True,)
-            ridgeCurrent = DraftGeomUtils.offset(profilCurrent["edge"],self.getPerpendicular(profilCurrent["vec"],profilCurrent["rot"],point[0].x))
-            point = DraftGeomUtils.findIntersection(ridgeCurrent,profilNext1["eave"],infinite1=True,infinite2=True,)
+    def helperGable(self, profilCurr, profilOther, isBack):
+        if isBack:
+            i = 0
         else:
-            point = DraftGeomUtils.findIntersection(profilCurrent["ridge"],profilNext1["eaveD"],infinite1=True,infinite2=True,)
-        self.ptsPaneProject.append(FreeCAD.Vector(point[0]))
-
-    def backPignon(self, i):
-
-        import DraftGeomUtils
-        profilCurrent = self.findProfil(i)
-        profilBack1 = self.findProfil(i-1)
-        profilBack2 = self.findProfil(i-2)
-        pt1 = DraftGeomUtils.findIntersection(profilCurrent["edge"],profilBack1["eave"],infinite1=True,infinite2=True,)
-        pt2 = DraftGeomUtils.findIntersection(profilBack2["edge"],profilBack1["eave"],infinite1=True,infinite2=True,)
-        eaveWithoutOverhang = DraftGeomUtils.edg(pt1[0],pt2[0])
-        if profilCurrent["run"]+profilBack2["run"] != eaveWithoutOverhang.Length :
-            points = [FreeCAD.Vector(0.0,0.0,0.0),]
-            points.append(FreeCAD.Vector(profilCurrent["run"],profilCurrent["height"],0.0))
-            rampantCurrent = DraftGeomUtils.edg(points[0],points[1])
-            points = [FreeCAD.Vector(eaveWithoutOverhang.Length,0.0,0.0),]
-            points.append(FreeCAD.Vector(eaveWithoutOverhang.Length-profilBack2["run"],profilBack2["height"],0.0))
-            rampantBack2 = DraftGeomUtils.edg(points[0],points[1])
-            point = DraftGeomUtils.findIntersection(rampantCurrent,rampantBack2,infinite1=True,infinite2=True,)
-            ridgeCurrent = DraftGeomUtils.offset(profilCurrent["edge"],self.getPerpendicular(profilCurrent["vec"],profilCurrent["rot"],point[0].x))
-            point = DraftGeomUtils.findIntersection(ridgeCurrent,profilBack1["eave"],infinite1=True,infinite2=True,)
+            i = 1
+        ptIntLst = find_inters(profilCurr["ridge"], profilOther["eaveDraft"])
+        if ptIntLst: # the edges of the roof segments are not parallel
+            ptProjLst = [ptIntLst[0]]
+        else: # the edges of the roof segments are parallel
+            ptProjLst = [profilCurr["ridge"].Vertexes[i].Point]
+        ptProjLst = ptProjLst + [profilCurr["eavePtLst"][i]]
+        if not isBack:
+            ptProjLst.reverse()
+        for ptProj in ptProjLst:
+            self.ptsPaneProject.append(ptProj)
+
+    def backGable(self, i):
+        profilCurr = self.findProfil(i)
+        profilBack = self.findProfil(i - 1)
+        self.helperGable(profilCurr, profilBack, isBack = True)
```

Comment:
The `helper_gable` function is not defined with default values.

To use this notation properly, it's better to set the default value in the definition.
```py
def helper_gable(profile_current, profile_back, is_back=True):
    ...
```

Then these three are equivalent
```py
helper_gable(profile_current, profile_back)
helper_gable(profile_current, profile_back, True)
helper_gable(profile_current, profile_back, is_back=True)
```

Reply:
The function is only used in the context of this program. I do not see much advantage.